libraries/common/subscriptions/helpers/handleLinks.js

```diff
@@ -143,11 +137,35 @@ export const isShopLink = (location) => {
   return true;
 };
 
+/**
+ * Check if the given pathname is a protector route.
+ * @param {string} location The location to check.
+ * @return {boolean}
+ */
+export const isProtector = location => Array.from(authRoutes.getAll().values()).includes(location);
+
+/**
+ * Finalizes the navigation to a legacy page or the in-app-browser.
+ * @param {string} historyAction he history action which was used to navigate.
+ */
+export const finializeNavigation = (historyAction) => {
+  const { pathname } = getCurrentRoute();
+
+  if (isProtector(pathname) && historyAction === ACTION_REPLACE) {
+    /**
+     * A replace action on a protector route indicates that the authentication was successful.
+     * So the protector route can be popped from the history stack.
```

Comment:
Same here.

Reply:
Like commented above, `protector` is the correct term, since it's about the route which protects a protected route.